e2e/installer_test.go

```diff
@@ -0,0 +1,91 @@
+package e2e
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/base32"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"testing"
+
+	"github.com/Azure/agentbakere2e/config"
+	"github.com/stretchr/testify/require"
+)
+
+// test node-bootstrapper binary without rebuilding VHD images.
+// it compiles the node-bootstrapper binary and uploads it to Azure Storage.
+// the runs the node-bootstrapper on the VM.
+func Test_ubuntu2204NodeBootstrapper(t *testing.T) {
+	if !config.Config.EnableNodeBootstrapperTest {
+		t.Skip("ENABLE_NODE_BOOTSTRAPPER_TEST is not set")
+	}
+	// TODO: figure out how to properly parallelize test, maybe move t.Parallel to the top of each test?
+	cluster, err := ClusterKubenet(context.TODO(), t)
+	require.NoError(t, err)
+	RunScenario(t, &Scenario{
+		Description: "Tests that a node using the Ubuntu 2204 VHD can be properly bootstrapped",
+		Config: Config{
+			//NodeBootstrappingType: SelfContained,
+			Cluster: ClusterKubenet,
+			VHD:     config.VHDUbuntu2204Gen2Containerd,
+			LiveVMValidators: []*LiveVMValidator{
+				containerdVersionValidator("1.7.20-1"),
+				runcVersionValidator("1.1.12-1"),
+			},
+			CSEOverride:       CSENodeBootstrapper(context.TODO(), t, cluster),
```

Comment:
what is context.TODO() ?

Reply:
Similar to `context.Background`, a context that never gets cancelled.
I replaced to a one with an expiration time.